coalahtml/tree.py

```diff
@@ -0,0 +1,81 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+    """
+    Node class that defines an individual node.
+    """
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+        self._id = id(self)
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(os.path.commonprefix(file_list))
+        self.root.abspath = self.root.path
+        self.fl_split = Tree._split_paths(file_list)
+        self._build(self.fl_split)
+
+    @staticmethod
+    def _split_paths(path_list):
```

Comment:
This probably shouldn't be a separate function. You use this only once and it's very simple one liner.